tasks/github_tasks.py

```diff
@@ -188,3 +189,41 @@ def get_token_from_app(_, app_id_env='GITHUB_APP_ID', pkey_env='GITHUB_KEY_B64')
     from .libs.common.github_api import GithubAPI
 
     GithubAPI.get_token_from_app(app_id_env, pkey_env)
+
+
+@task
+def deduce_team_label(_, changed_files=[]):
+    """
+    Print the github team label name if a single team can
+    be deduced from the changed files
+    """
+    def get_team() -> str | None:
+        from codeowners import CodeOwners
+
+        with open('.github/CODEOWNERS') as f:
+            codeowners = CodeOwners(f.read())
+
+        global_team = None
+        for file in changed_files:
+            owners = [name for (kind, name) in codeowners.of(file) if kind == 'TEAM']
+            if len(owners) != 1:
+                # Multiple / no owners case
+                return
+            else:
+                file_team = owners[0]
+                if global_team is not None and file_team != global_team:
+                    # Multiple owners case
+                    return
+                else:
+                    global_team = file_team
+
+        return global_team
+
+    # Find team
+    team = get_team()
+    assert team is not None, 'No team or multiple teams found'
+    assert team.startswith('@DataDog/'), f'Unknown team {team}'
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Discussed offline earlier: this would be caught by `gh` anyway, wouldn't it?

Reply:
Yes you're right !